pyo3-macros-backend/src/attributes.rs

```diff
@@ -0,0 +1,96 @@
+use syn::{
+    parse::{Parse, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    token::Comma,
+    Attribute, ExprPath, Ident, LitStr, Result, Token,
+};
+
+pub mod kw {
+    syn::custom_keyword!(annotation);
+    syn::custom_keyword!(attribute);
+    syn::custom_keyword!(from_py_with);
+    syn::custom_keyword!(item);
+    syn::custom_keyword!(pass_module);
+    syn::custom_keyword!(name);
+    syn::custom_keyword!(signature);
+    syn::custom_keyword!(transparent);
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct FromPyWithAttribute(pub ExprPath);
+
+impl Parse for FromPyWithAttribute {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let _: kw::from_py_with = input.parse()?;
+        let _: Token![=] = input.parse()?;
+        let string_literal: LitStr = input.parse()?;
+        string_literal.parse().map(FromPyWithAttribute)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct NameAttribute(pub Ident);
+
+impl Parse for NameAttribute {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let _: kw::name = input.parse()?;
+        let _: Token![=] = input.parse()?;
+        let string_literal: LitStr = input.parse()?;
```

Comment:
I'm going to keep them separate, because in the future I might want to change it to

```rust
#[derive(Clone, Debug, PartialEq)]
pub struct NameAttribute {
    pub name: kw::name,
    pub eq: Token![=],
    pub ident: Ident,
}
```

and keep all the contents when parsing - this is quite common with `syn` so that spans can be built nicely.